From 7fe5b0ae2e86405d32cab502130e804c7ffdf58b Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:13:04 -0400
Subject: [ethernet] i40evf: do not re-arm watchdog after remove

Message-id: <1409663591-32531-142-git-send-email-sassmann@redhat.com>
Patchwork-id: 90617
O-Subject: [RHEL7.1 PATCH 141/148] i40evf: do not re-arm watchdog after remove
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Ashish Shah <ashish.n.shah@intel.com>

Add in an adapter state check to prevent re-arming watchdog timer after
i40evf_remove has been called and timer has been deleted.

Change-ID: I636ba7c6322be8cbf053231959f90c0a2d8d803a
Signed-off-by: Ashish Shah <ashish.n.shah@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit d3e2edb70ecf877b4e5f42314449fc648b18627b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_main.c b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
index 4c01079..2aca9cf 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_main.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
@@ -1403,6 +1403,8 @@ static void i40evf_watchdog_task(struct work_struct *work)
 watchdog_done:
  clear_bit(__I40EVF_IN_CRITICAL_TASK, &adapter->crit_section);
 restart_watchdog:
+ if (adapter->state == __I40EVF_REMOVE)
+  return;
  if (adapter->aq_required)
   mod_timer(&adapter->watchdog_timer,
      jiffies + msecs_to_jiffies(20));
-- 
1.7.1